### DISCUSSION TITLE
some question when i invoke my function

### DISCUSSION THREAD

**Discussion by P001water** [pallets/click#2372] [Q&A]
`import sys, click, os

def two_function(DataFile="test"):
    print("test")
    print(DataFile)
    

@click.command()
@click.option('-f', '--file', required=True, help="file")
def tshark_do(file):
    print(file)
    

if __name__ == "__main__":
    tshark_do()
    print(1)
    two_function()
    print(2)`

seemd that function `print(1)`, `two_function()`, `print(2)` can't be invoked , is there a good idea to solve it if i want to invoke them?

**Comment by jules43** [comment]
Can you explain what you're trying to achieve? In your example you could call two_function() from within tshark_do(file), but I presume there's a reason why that doesn't solve your problem.

**Reply by P001water** [reply]
thank you bro ,  i run that with __python demo.py -f file__ , it just output __file__ , here' s a question that why it didn't output 
- 1 
- test
- test
- 2
?  seemd that functions `print(1)`, `two_function()`, `print(2)` were not called ,  If I want it to output as above, what can I do
